### PR TITLE
feat: add Edit profile page UI and basic functionality

### PR DESCRIPTION
**Pull request by anette1983**
## Describe your changes

- I added Edit profile page UI and basic functionality

Знімок екрана 2023-11-29 195828
Знімок екрана 2023-11-29 195924
Знімок екрана 2023-11-29 195958
Знімок екрана 2023-11-29 200105
Знімок екрана 2023-11-29 200148

## Issue ticket code (and/or) and link

- [Link to JIRA ticket](https://testhomework4.atlassian.net/jira/software/projects/AG/boards/6?selectedIssue=AG-119)

### **General**

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [ ] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [ ] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [ ] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [ ] Date and time formats are on the constants
- [ ] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [ ] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Frontend

- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the constants
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [ ] Imports are absolute
- [x] Attach a screenshot if PR has visual changes.

### CODE REVIEW

File: src/app/[locale]/edit-profile/page.tsx [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+import UserEditView from 'sections/user/view/user-edit-view';
+
+
+function EditProfilePage() {
+  return <UserEditView />;
```
Comment (AKutusenko) [inline src/app/[locale]/edit-profile/page.tsx:5]: add function return type

File: src/components/hook-form/rhf-text-area.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
+import { useFormContext, Controller } from 'react-hook-form';
+import TextField, { TextFieldProps } from '@mui/material/TextField';
+
+type Props = TextFieldProps & {
+  name: string;
+};
+
+export default function RHFTextArea({ name, helperText, type, ...other }: Props) {
+  const { control } = useFormContext();
```
Comment (AKutusenko) [inline src/components/hook-form/rhf-text-area.tsx:9]: same

File: src/sections/user/user-edit-settings.tsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+import CardHeader from '@mui/material/CardHeader';
+import Iconify from 'components/iconify';
+
+interface RenderProps {
+  checked: boolean;
+  handleChange: (event: ChangeEvent<HTMLInputElement>) => void;
+  theme: Theme;
+}
+
+const RenderSettings = ({ checked, handleChange, theme }: RenderProps) => {
```
Comment (AKutusenko) [inline src/sections/user/user-edit-settings.tsx]: same

File: src/sections/user/user-edit-settings.tsx [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+              onChange={handleChange}
+              inputProps={{ 'aria-label': 'controlled' }}
+            />
+          </Box>
+        </Stack>
+      </Stack>
+    </Card>
+  );
+};
+export function ProfileSettings() {
```
Comment (AKutusenko) [inline src/sections/user/user-edit-settings.tsx]: same

File: src/sections/user/user-edit-settings.tsx [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+  const [checked, setChecked] = useState<boolean>(true);
+  const theme = useTheme();
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
```
Comment (AKutusenko) [inline src/sections/user/user-edit-settings.tsx]: same

File: src/sections/user/user-new-edit-form.tsx [hunk 6]
```diff
@@ -0,0 +1,228 @@
...
+  currentUser?: IUserEditItem;
+};
+
+function getRoles() {
+  const roles = ['Agent', 'Agency'];
+
+  return roles;
+}
+
+export default function UserNewEditForm({ currentUser }: Props) {
```
Comment (AKutusenko) [inline src/sections/user/user-new-edit-form.tsx]: same

File: src/components/hook-form/rhf-text-area.tsx [hunk 7]
```diff
@@ -0,0 +1,42 @@
+import { useFormContext, Controller } from 'react-hook-form';
+import TextField, { TextFieldProps } from '@mui/material/TextField';
+
+type Props = TextFieldProps & {
+  name: string;
+};
+
+export default function RHFTextArea({ name, helperText, type, ...other }: Props) {
```
Comment (canterbery) [inline src/components/hook-form/rhf-text-area.tsx]: return type

File: src/sections/user/user-new-edit-form.tsx [hunk 8]
```diff
@@ -0,0 +1,228 @@
...
+  });
+
+  const {
+    reset,
+    setValue,
+    handleSubmit,
+    formState: { isSubmitting },
+  } = methods;
+
+  const onSubmit = handleSubmit(async (data) => {
```
Comment (AKutusenko) [inline src/sections/user/user-new-edit-form.tsx]: same

File: src/sections/user/view/user-edit-view.tsx [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+import { useMockedUser } from 'hooks/use-mocked-user';
+import { _userList } from '_mock';
+import { paths } from 'routes/paths';
+import { useSettingsContext } from 'components/settings';
+import CustomBreadcrumbs from 'components/custom-breadcrumbs';
+import { SnackbarProvider } from 'components/snackbar';
+import UserNewEditForm from '../user-new-edit-form';
+import ProfileSettings from '../user-edit-settings';
+
+export default function UserEditView() {
```
Comment (AKutusenko) [inline src/sections/user/view/user-edit-view.tsx]: same

File: src/sections/user/user-edit-settings.tsx (hunk 3 above)
Comment (canterbery) [inline src/sections/user/user-edit-settings.tsx:18]: same

File: src/sections/user/user-new-edit-form.tsx [hunk 10]
```diff
@@ -0,0 +1,228 @@
...
+import Iconify from 'components/iconify';
+import { useSnackbar } from 'components/snackbar';
+import FormProvider, { RHFTextField, RHFUploadAvatar, RHFAutocomplete } from 'components/hook-form';
+import RHFTextArea from 'components/hook-form/rhf-text-area';
+
+type Props = {
+  currentUser?: IUserEditItem;
+};
+
+function getRoles() {
```
Comment (canterbery) [inline src/sections/user/user-new-edit-form.tsx]: return type

**Review by AKutusenko — APPROVED** [review]

**Review by lyudad — APPROVED** [review]

**Pull request merged**